test/extended/builds/no_outputname.go

```diff
@@ -25,11 +25,21 @@ var _ = g.Describe("[builds][Conformance] build without output image", func() {
 			o.Expect(err).NotTo(o.HaveOccurred())
 
 			g.By("expecting build to pass without an output image reference specified")
-			out, err := oc.Run("start-build").Args("test-docker", "--follow", "--wait").Output()
+			err = oc.Run("start-build").Args("test-docker").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("waiting for the build to complete")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), "test-docker-1", exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
 			if err != nil {
-				fmt.Fprintln(g.GinkgoWriter, out)
+				exutil.DumpBuildLogs("test-docker", oc)
 			}
 			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("verifying the build test-docker-1 output")
+			out, err := oc.Run("logs").Args("-f", "bc/test-docker").Output()
+			fmt.Fprintf(g.GinkgoWriter, "\n: build logs: %s\n", out)
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
why are we retrieving/printing the logs here, when we already handled dumping them on an error above?